### PR TITLE
Dependencies: arena-core latest version

### PR DESCRIPTION
**Pull request by SteRiccio**

### CODE REVIEW

File: test/unit/tests/002recordExpressionParser.test.js [hunk 1]
```diff
@@ -168,8 +166,8 @@ describe('RecordExpressionParser Test', () => {
     // global objects (String)
     { q: 'String.fromCharCode(65, 66, 67)', r: 'ABC' },
     // global objects (unknown objects/functions)
-    { q: 'Invalid.func(1)', e: new SystemError(Validation.messageKeys.expressions.unableToFindNode) },
-    { q: 'Math.unknownFunc(1)', e: new SystemError('undefinedFunction') },
+    { q: 'Invalid.func(1)', e: new SystemError('expression.identifierNotFound') },
+    { q: 'Math.unknownFunc(1)', e: new SystemError('expression.identifierNotFound') },
```
Comment (SteRiccio) [inline test/unit/tests/002recordExpressionParser.test.js:170]: should there be some kind of message keys object for these keys? should it go to arena-core?

Comment (ramirobg94) [reply test/unit/tests/002recordExpressionParser.test.js:169]: yes we should add this

**Review by ramirobg94 — COMMENTED** [review]
What is the way to get the dependent nodes of a node?
and the way to validate a ndoe?

File: test/unit/tests/003SqlUtil.test.js [hunk 2]
```diff
@@ -83,14 +82,14 @@ const badExpressions = [
 describe('dataFilter test', () => {
   goodExpressions.forEach(({ q, r }) => {
     test(q, () => {
-      const ps = Expression.toSql(jsep(q))
+      const ps = Expression.toSql(Expression.fromString(q))
```
[the remark this answers is not in the document]
Comment (ramirobg94) [reply test/unit/tests/003SqlUtil.test.js:85]: I like the JsonToSql,  I think that the format is AST but not sure,

File: core/expressionParser/expression.js [hunk 3]
```diff
@@ -41,7 +42,7 @@ export const fromString = (string, exprMode = modes.json) => {
           R.replace(/<==/g, '<=')
```
Comment (ramirobg94) [inline core/expressionParser/expression.js:42]: maybe we can remove R dependencies

File: test/unit/tests/003SqlUtil.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (SteRiccio) [reply test/unit/tests/003SqlUtil.test.js:85]: so should we add a jsonToSql function to Expression?

File: test/unit/tests/002recordExpressionParser.test.js (hunk 1 above)
Comment (SteRiccio) [reply test/unit/tests/002recordExpressionParser.test.js:169]: ok... should it have nested objects? like: 
systemErrorKeys: {
  expression: {
    identifierNotFound
  }
}
or should it be just an enum?
SystemErrorKeys {
  ExpressionIdentifierNotFound,
  ...
}

File: test/unit/tests/003SqlUtil.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (SteRiccio) [reply test/unit/tests/003SqlUtil.test.js:85]: ah no, I mean a function fromStringToSql? that format is AST, right, the one passed to the toSql function I mean, but the string passed to fromString is just the draft expression

File: core/record/recordExpressionParser/recordExpressionParser.js [hunk 4]
```diff
@@ -1,51 +1,16 @@
-import * as R from 'ramda'
+import { RecordExpressionEvaluator } from '@openforis/arena-core'
```
Comment (ramirobg94) [inline core/record/recordExpressionParser/recordExpressionParser.js:1]: did we move that?

File: core/survey/_survey/surveyDependencies.js [hunk 5]
```diff
@@ -121,47 +62,29 @@ export const isNodeDefDependentOn = (nodeDefUuid, nodeDefSourceUuid) => (survey)
...
-  let dependencyGraphUpdated
-  if (dependencyType) {
-    dependencyGraphUpdated = R.dissocPath([dependencyType, nodeDefUuid])(dependencyGraph)
-  } else {
-    const dependencyTypes = Object.keys(dependencyGraph)
-    dependencyGraphUpdated = dependencyTypes.reduce(
-      (dependencyGraphAcc, type) => R.dissocPath([type, nodeDefUuid])(dependencyGraphAcc),
-      dependencyGraph
-    )
+export const removeNodeDefDependencies =
```
Comment (ramirobg94) [inline core/survey/_survey/surveyDependencies.js:75]: is this somethng to move?

**Review by ramirobg94 — APPROVED** [review]

**Pull request merged**